lib/ansible/module_utils/basic.py

```diff
@@ -1993,7 +1993,7 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                             try:
                                 try:
                                     out_dest = open(b_dest, 'wb')
-                                    in_src = open(b_src, 'rb')
+                                    in_src = open(b_tmp_dest_name, 'rb')
```

Comment:
Not sure why but at this point in code, b_src refers to a non existing file when b_tmp_dest_name refers to a file that exists and has correct data inside.

When running with b_src we have this error:

```
TASK [insert 127.0.0.2 line in /etc/hosts] *************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: UnboundLocalError: local variable 'in_src' referenced before assignment
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_WvGe4f/ansible_module_lineinfile.py\", line 425, in <module>\n    main()\n  File \"/tmp/ansible_WvGe4f/ansible_module_lineinfile.py\", line 417, in main\n    ins_aft, ins_bef, create, backup, backrefs)\n  File \"/tmp/ansible_WvGe4f/ansible_module_lineinfile.py\", line 298, in present\n    write_changes(module, b_lines, dest)\n  File \"/tmp/ansible_WvGe4f/ansible_module_lineinfile.py\", line 175, in write_changes\n    unsafe_writes=module.params['unsafe_writes'])\n  File \"/tmp/ansible_WvGe4f/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2002, in atomic_move\nUnboundLocalError: local variable 'in_src' referenced before assignment\n", "module_stdout": "", "msg": "MODULE FAILURE"}
        to retry, use: --limit @/tmp/ansible/etc_host.retry
```

This is an error raised by the incorrect handling of file not found exception. Removing this line handler (line 2002) we can get to the actual issue:

```
TASK [insert 127.0.0.2 line in /etc/hosts] *************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Could not write data to file (/etc/hosts) from (/tmp/tmpll43Kj): [Errno 2] No such file or directory: '/tmp/tmpll43Kj'"}
        to retry, use: --limit @/tmp/ansible/etc_host.retry
```